### PR TITLE
Ensure option types are publicly available

### PR DESCRIPTION
**Pull request by leo**
At the moment, as mentioned on Slack by @colodenn, the type of a `Transaction` is incomplete, because our build tool removes the types of internal methods, which that type relies on.

By splitting out the configuration type into a dedicated interface, we are preventing the build tool from stripping it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,7 +103,24 @@ new Transaction(queries, {
...
+  // a list (array) of values, which are inherently not prone to conflicts.
+  //
+  // If the driver being used instead returns an object for every row, the driver must
+  // ensure the uniqueness of every key in that object, which means prefixing duplicated
+  // column names with the name of the respective table, if multiple tables are joined.
+  // Drivers that return objects for rows offer this behavior as an option that is
+  // usually called "expand columns". If the driver being used does not offer such an
+  // option, you can instead activate the option in the compiler, which results in longer
+  // SQL statements because any duplicated column name is aliased.
+  expandColumns: true
```
Comment (leo) [inline README.md:123]: This is a new option that is not yet implemented, but will be with a separate PR.

**Review by colodenn — APPROVED** [review]
